### PR TITLE
Add secret and endpoint config to readme

### PR DESCRIPTION
**Pull request by pnwpedro**
## Problem

The readme is missing details on configuring the client.

## Solution

* Add the most critical configuration guidance, `secret` and `endpoint`. The rest can be found by looking at the init signature.
* Fix a few grammar issues.

## Testing

N/A

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -36,14 +36,24 @@ The following versions of Python are supported:
...
 .. code-block:: python
 
     from fauna import fql
     from fauna.client import Client
     from fauna.errors import FaunaException
 
-    client = Client()
+    # Assumes you have retrieved a secret from a secret store and set it to `my_secret`.
+    # As a best practice, don't store your secret directly in your code.
+    client = Client(secret=my_secret)
```
Comment (cynicaljoy) [inline README.rst:56]: I think we only demonstrate the use of `client = Client()` to encourage the use of the env var.

File: README.rst [hunk 2]
```diff
@@ -36,14 +36,24 @@ The following versions of Python are supported:
 
 Basic Usage
 -------------
+You can expect a ``Client`` instance to have reasonable defaults, like the Fauna endpoint ``db.fauna.com`` and a global HTTP client, but you will always need to supply a secret.
```
Comment (cynicaljoy) [inline README.rst:39]: this reads as if arg `secret=""` is required

Comment (pnwpedro) [reply README.rst:39]: The next line says how. What do you propose?

Comment (pnwpedro) [reply README.rst:39]: how about 

```suggestion
You can expect a ``Client`` instance to have reasonable defaults, like the Fauna endpoint ``db.fauna.com`` and a global HTTP client, but you will always need to configure a secret.
```

File: README.rst (hunk 1 above)
Comment (pnwpedro) [reply README.rst:56]: Maybe this?

```suggestion
    # As a best practice, don't store your secret directly in your code.
    # Either set the FAUNA_SECRET, or retrieve it from a secret store.
    client = Client(secret=my_secret)
```

File: README.rst [hunk 3]
```diff
@@ -36,14 +36,25 @@ The following versions of Python are supported:
...
 .. code-block:: python
 
     from fauna import fql
     from fauna.client import Client
     from fauna.errors import FaunaException
 
-    client = Client()
+    # Assumes you have retrieved a secret from a secret store and set it to `my_secret`.
+    # As a best practice, don't store your secret directly in your code.
+    # Either set the FAUNA_SECRET, or retrieve it from a secret store.
```
Comment (pnwpedro) [inline README.rst:56]:
```suggestion
    # Either set the FAUNA_SECRET env variable, or retrieve it from a secret store.
```

**Review by cynicaljoy — APPROVED** [review]

File: README.rst [hunk 4]
```diff
@@ -13,7 +13,7 @@ Python driver for `Fauna <https://fauna.com>`_.
     It operates with the Fauna database service via an API which is also in
     beta release, and is not recommended for production use. This driver is
```
[the remark this answers is not in the document]
Comment (pnwpedro) [reply README.rst:14]:
```suggestion
    beta release and is not recommended for production use. This driver is
```

File: README.rst [hunk 5]
```diff
@@ -36,14 +36,25 @@ The following versions of Python are supported:
...
 .. code-block:: python
 
     from fauna import fql
     from fauna.client import Client
     from fauna.errors import FaunaException
 
-    client = Client()
+    # Assumes you have retrieved a secret from a secret store and set it to `my_secret`.
+    # As a best practice, don't store your secret directly in your code.
+    # Either set the FAUNA_SECRET env variable, or retrieve it from a secret store.
```
[the remark this answers is not in the document]
Comment (pnwpedro) [reply README.rst:55]:
```suggestion
    # Either set the FAUNA_SECRET env variable or retrieve it from a secret store.
```

**Pull request merged**